accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,333 @@
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The `dotnet package update` feature is designed to enhance the user experience by enabling .NET developers to address security vulnerabilities, deprecated packages, and outdated versions of their NuGet dependencies with a single command.
+This improvement will, in turn, bolster the overall security and reliability of .NET applications. The feature will build upon the existing NuGet Audit functionality.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Even though .NET developers now have the opportunity to learn about vulnerabilities and deprecated dependencies in their NuGet packages, addressing these issues remains a significant challenge. It requires a lot of manual work and knowledge about newer versions, which can be quite daunting. When the solution to these problems takes time, developers are less likely to act on it immediately, leading to these upgrades being postponed or never addressed. Additionally, developers might not always use the most efficient ways of resolving these issues, especially when dealing with multiple nested transitive dependencies. This is why developers will benefit greatly from a single command that attempts to automate all the work and update all the packages, efficiently eliminating vulnerabilities and deprecated versions.
+
+## User experience
+
+### Updating all packages to the latest vesioin
+
+When users want to ensure their NuGet dependencies are up to date, they can run a command in .NET CLI `dotnet package update`.
+
+```CLI
+C:\> dotnet package update
+
+Fixing outdated packages in ContosoUniversity.sln
+ Updated UmbracoForms 8.4.1 to 8.7.1
+ Updated Newtonsoft.Json 11.0.1 to 13.0.3
+    
+Updated 2 packages in 36 scanned packages.
+```
+
+### Fixing vulnerabilities
+
+When users run `dotnet build` or `dotnet restore` commands in CLI, if they see any warnings related to vulnerabilities in their project’s NuGet dependencies, they can run `dotnet package update --vulnerable` CLI command to fix all the mentioned issues.
+
+In the future this can be extended to fix deprecated versions.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### dotnet audit
+
+The `dotnet audit` command will scan your project and solution's dependencies and look for known vulnerabilities, deprecations, and outdated packages. If any vulnerabilities, deprecations, or outdated packages are found, a remediation will be calculated to then `fix` which will be applied to the resulting package graph.
+
+Example:
+
+![dotnet audit](/meta/resources/dotnetaudit/dotnetaudit.png)
+
+```
+$ dotnet audit
```

Comment:
This effective combines `dotnet list package --outdated`, `dotnet list package --vulnerable`,  and `dotnet list package --deprecated`. 

I'm not saying the output should be the same, but it's worth considering them as inspiration. 
I think in general, consistency across the different commands helps.